rsmt2d_test.go

```diff
@@ -157,3 +158,57 @@ func TestEdsRepairTwice(t *testing.T) {
 		})
 	}
 }
+
+// TestRepairWithOneQuarterPopulated is motivated by a use case from
+// celestia-node. It verifies that a new EDS can be populated via SetCell. After
+// enough chunks have been populated, it verifies that the EDS can be repaired.
+// After the EDS is repaired, the test verifies that data in a repaired cell
+// matches the expected data.
+func TestRepairWithOneQuarterPopulated(t *testing.T) {
+	edsWidth := 4
+	chunkSize := 512
+
+	exampleEds := createExampleEds(t, chunkSize)
+
+	eds, err := rsmt2d.NewExtendedDataSquare(rsmt2d.NewLeoRSCodec(), rsmt2d.NewDefaultTree, uint(edsWidth), uint(chunkSize))
+	require.NoError(t, err)
+
+	// Populate EDS with 1/4 of chunks using SetCell
+	err = eds.SetCell(0, 0, exampleEds.GetCell(0, 0))
+	require.NoError(t, err)
+	err = eds.SetCell(0, 1, exampleEds.GetCell(0, 1))
+	require.NoError(t, err)
+	err = eds.SetCell(1, 0, exampleEds.GetCell(1, 0))
+	require.NoError(t, err)
+	err = eds.SetCell(1, 1, exampleEds.GetCell(1, 1))
+	require.NoError(t, err)
+
+	// Verify that an unpopulated cell returns nil
+	assert.Nil(t, eds.GetCell(3, 3))
+
+	rowRoots, err := exampleEds.RowRoots()
+	require.NoError(t, err)
+	colRoots, err := exampleEds.ColRoots()
+	require.NoError(t, err)
+
+	// Repair the EDS
+	err = eds.Repair(rowRoots, colRoots)
+	assert.NoError(t, err)
+
+	assert.Equal(t, exampleEds.GetCell(3, 3), eds.GetCell(3, 3))
```

Comment:
[Question] I'd suggest checking all the cells for equality (like through a loop).

Reply:
Resolved via 11bb1c0a7c3531e6813fca015496234ba1f47988